### PR TITLE
Fix consistency get_fact() for NXOS and NXOS_SSH drivers

### PR DESCRIPTION
**Pull request by FloLaco**
Fix bug #477 

I've edited the result for the `NXOS_SSH` driver to return the same formating regarting to the `NXOS` driver.
Change I made : 
- The hostname does not return the FQDN anymore
- The model does include the full name until `(brackets for sup module)`
- Consistency interface name using canonical name (Ethernet, port-channel, loopback)
- Fix an empty version for some model

### CODE REVIEW

File: test/nxos_ssh/mocked_data/test_get_facts/missing_domain/expected_result.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "hostname": "nxos1",
     "fqdn": "",
-    "model": "NX-OSv", 
+    "model": "NX-OSv Chassis",
```
Comment (mirceaulinic) [inline test/nxos_ssh/mocked_data/test_get_facts/missing_domain/expected_result.json]: To be honest, I'm not sure about this, but looking above at `"model": "Nexus7000 C7009"` I understand your reasoning.

File: napalm/nxos_ssh/nxos_ssh.py [hunk 2]
```diff
@@ -785,7 +791,15 @@ def get_facts(self):
             if line.startswith(' ') or line.startswith('-') or line.startswith('Port '):
                 continue
             interface = line.split()[0]
-            interface_list.append(interface)
+            # Return canonical interface name
+            if re.search(r'^Eth', interface):
```
Comment (mirceaulinic) [inline napalm/nxos_ssh/nxos_ssh.py]: I like having this consistent.

**Comment by mirceaulinic** [comment]
If we are going to accept this, I think we'll get it into a major release only (scheduled for 2.4.0 if everything works as planned).

File: test/nxos_ssh/mocked_data/test_get_facts/missing_domain/expected_result.json (hunk 1 above)
Comment (mirceaulinic) [reply test/nxos_ssh/mocked_data/test_get_facts/missing_domain/expected_result.json]: Oh I just realised @ktbyers has the same changes in #476.

Comment (Rian83) [reply test/nxos_ssh/mocked_data/test_get_facts/missing_domain/expected_result.json]: @mirceaulinic about model, I take into reference the nxapi output. I've installed a titanium switch (nxos virtual) and try the get_facts on both.

Comment (Rian83) [reply test/nxos_ssh/mocked_data/test_get_facts/missing_domain/expected_result.json]: Oh yes, @ktbyers didn't link his PR to his issue so I didn't know about it oups.
But it doesn't work the same way, specially for the hostname

**Comment by ktbyers** [comment]
I don't think it should go in today. I need to review what I did in that previous PR and there was an issue on that PR that I created that needed fixed.

For any consistency issues we need to look at napalm-base and decide how it should be implemented and then document it there. For example `hostname` and whether it should include the domain-name or not.

In general, I think nxos_ssh is probably a more trustworthy source of truth than nx-api. It was implemented more recently and likely has more test data.

For any interface name expansion we should be using the `canonical_name` support that @itdependsnetworks added in the recent hackathon.

**Comment by mirceaulinic** [comment]
I'm checking in here:

> For example hostname and whether it should include the domain-name or not.

I think it should be like that, yes.

> In general, I think nxos_ssh is probably a more trustworthy source of truth than nx-api. It was implemented more recently and likely has more test data.

+1

> For any interface name expansion we should be using the canonical_name support that @itdependsnetworks added in the recent hackathon.

I agree. @Rian83 are you okay with that?

**Comment by dbarrosop** [comment]
> For example hostname and whether it should include the domain-name or not

I don't think it should. hostname != fqdn.

**Comment by mirceaulinic** [comment]
I just realised my answer is confusing. I also vote +1 for "I don't think it should. hostname != fqdn." - just to clarify what I meant above.

**Comment by Rian83** [comment]
> In general, I think nxos_ssh is probably a more trustworthy source of truth than nx-api. It was implemented more recently and likely has more test data.

Totally agree with that.

@mirceaulinic about canonical interface, yes we should use the it because it's more trustful for the futur. And there is so possibility about the contracted form ("eth", "e", "ether", etc ..)
And more and more i'm working latest nexus 9000, and more I discover some new interface name, like BDI, etc. We definitly should use the canonical name ("Ethernet") instead of the contracted name ("Eth" or E).

**Comment by itdependsnetworks** [comment]
Can you make use of: https://github.com/napalm-automation/napalm/blob/68185ac9cec5350b2d461b00ca03563b4ddfc428/napalm/base/helpers.py#L271

**Comment by FloLaco** [comment]
@itdependsnetworks what's the best practice for adding  some elements in your `base_interfaces = { `

**Comment by mirceaulinic** [comment]
Hey @FloLaco can you please elaborate? If you have a list of interfaces and you're unsure how to map them into `base_interfaces`, just share them here and we or @itdependsnetworks will be able to advise.

**Comment by itdependsnetworks** [comment]
@FloLaco you should be able to just add to the at mapping. For each possible short name, create a key value pair in base_interfaces, and in reverse_mapping pick a standard short name to reverse to.

**Comment by FloLaco** [comment]
I'm using the `canonical_interface_name()` function, but for port-channel, it's returning `PortChannel17` while in a Cisco syntax, it is more like `Port-Channel17`. Which format should we use ?

**Comment by dbarrosop** [comment]
@itdependsnetworks would you mind answering that question?

**Comment by FloLaco** [comment]
Ok i'll fix it.

**Comment by FloLaco** [comment]
@ktbyers I've make the changes about the `canonical_interface_name()` utilization

**Comment by ktbyers** [comment]
This PR is superceded by: https://github.com/napalm-automation/napalm/pull/674

**Pull request closed without merging**